NuRadioReco/examples/PhasedArray/SNR_curves/T02RunSNR_RNOG.py

```diff
@@ -0,0 +1,280 @@
+import NuRadioReco.modules.efieldToVoltageConverter
+import NuRadioReco.modules.trigger.simpleThreshold
+import NuRadioReco.modules.phasedarray.triggerSimulator
+import NuRadioReco.modules.channelResampler
+import NuRadioReco.modules.channelBandPassFilter
+import NuRadioReco.modules.channelGenericNoiseAdder
+from NuRadioReco.utilities import units
+from NuRadioReco.modules.base import module
+
+logger = module.setup_logger(level=logging.WARNING)
+
+# initialize detector sim modules
+efieldToVoltageConverter = NuRadioReco.modules.efieldToVoltageConverter.efieldToVoltageConverter()
+efieldToVoltageConverter.begin(debug=False)
+triggerSimulator = NuRadioReco.modules.phasedarray.triggerSimulator.triggerSimulator()
+channelResampler = NuRadioReco.modules.channelResampler.channelResampler()
+channelBandPassFilter = NuRadioReco.modules.channelBandPassFilter.channelBandPassFilter()
+channelGenericNoiseAdder = NuRadioReco.modules.channelGenericNoiseAdder.channelGenericNoiseAdder()
+thresholdSimulator = NuRadioReco.modules.trigger.simpleThreshold.triggerSimulator()
+
+# 4 channel, 2x sampling, fft upsampling, Half Window
+#  100 Hz -> 2.63
+#  10 Hz -> 2.75
+#  1 Hz -> 2.86
+
+# 8 channel, 4x sampling, fft upsampling, Half Window
+#  100 Hz -> 2.67
+#  10 Hz -> 2.78
+#  1 Hz -> 2.89
+
+# Single antenna
+#  100 Hz -> 2.02
+#  10 Hz -> 2.12
+#  1 Hz -> 2.21
+
+n_channels = 8
+main_low_angle = np.deg2rad(-59.55)
+main_high_angle = np.deg2rad(59.55)
+
+N = 31
+SNRs = (np.linspace(0.5, 4.0, N))
+SNRtriggered = np.zeros(N)
+
+channels = []
+
+min_freq = 0.0 * units.MHz
+max_freq = 250.0 * units.MHz
+fff = np.linspace(min_freq, max_freq, 10000)
+filt1_highres = channelBandPassFilter.get_filter(fff, 0, 0, None, passband=[0, 240 * units.MHz], filter_type="cheby1", order=9, rp=.1)
+filt2_highres = channelBandPassFilter.get_filter(fff, 0, 0, None, passband=[80 * units.MHz, 230 * units.MHz], filter_type="cheby1", order=4, rp=.1)
+filt_highres = filt1_highres * filt2_highres
+bandwidth = np.trapz(np.abs(filt_highres) ** 2, fff)
+Vrms_ratio = np.sqrt(bandwidth / (max_freq - min_freq))
+
+if(n_channels == 4):
+    upsampling_factor = 2
+    phasing_angles = np.arcsin(np.linspace(np.sin(main_low_angle), np.sin(main_high_angle), 11))
+    channels = np.arange(4, 8)
+    threshold = 2.63
+    phase = True
+elif(n_channels == 8):
+    upsampling_factor = 4
+    phasing_angles = np.arcsin(np.linspace(np.sin(main_low_angle), np.sin(main_high_angle), 21))
+    channels = None
+    threshold = 2.67
+    phase = True
+elif(n_channels == 1):
+    upsampling_factor = 1
+    phasing_angles = []
+    channels = [4]
+    threshold = 2.02
+    phase = False
+else:
+    print("wrong n_channels!")
+    exit()
+
+def count_events():
+    count_events.events += 1
+count_events.events = 0
+
+class mySimulation(simulation.simulation):
+
+    def _detector_simulation_filter_amp(self, evt, station, det):
+        channelBandPassFilter.run(evt, station, det,
```

Comment:
inconsistent definition of filter. Should be the same as in `detector_simulation_part2`